ErrorHandling.js

```diff
@@ -0,0 +1,98 @@
+class CarDamageException extends Error {}
+class InsuranceClaimException extends Error {}
+class InjuryException extends Error {}
+class AirbagDeployedException extends Error {}
+class GasLeakException extends Error {}
+
+try {
+    throw new CarDamageException('Car got damaged due to the accident!');
+} catch (error) {
+    if (error instanceof CarDamageException) {
+        callMechanic();
+    }
+} finally {
+    carDamageHandlingCompleted();
+}
+
+try {
+    throw new InsuranceClaimException(
+        'File insurance claim after the accident!',
+    );
+} catch (error) {
+    if (error instanceof InsuranceClaimException) {
+        fileInsuranceClaim();
+    }
+} finally {
+    insuranceClaimHandlingCompleted();
+}
+
+try {
+    throw new InjuryException('Someone is injured in the accident!');
+} catch (error) {
+    if (error instanceof InjuryException) {
+        provideMedicalAssistance();
+    }
+} finally {
+    medicalAssistanceProvided();
+}
+
+try {
+    throw new AirbagDeployedException('Airbag deployed during the accident!');
```

Comment:
Same goes with this, Is **AirbagDeployedException** an exception or a solution/action?

Reply:
I think it should have a better name like maybe **AirbagDeploymentFailedException** or something so that it tells us exactly why this is an exception.

Reply:
The "AirbagDeploymentFailedException" itself is not a solution or action after an accident. It is an exception that represents a problem, specifically related to the process of opening the airbags after the accident